frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/VerifyMfaCodeHandler.java

```diff
@@ -171,7 +172,18 @@ public APIGatewayProxyResponseEvent handleRequestWithUserContext(
         var journeyType = codeRequest.getJourneyType();
         Optional<UserProfile> userProfileMaybe = userContext.getUserProfile();
         UserProfile userProfile = userProfileMaybe.orElse(null);
-        var maybeRpPairwiseId = getRpPairwiseId(userProfile, userContext.getClient());
+        ClientRegistry client;
+        try {
+            client =
+                    userContext
+                            .getClient()
+                            .orElseThrow(
+                                    () -> new ClientNotFoundException(userContext.getSession()));
+        } catch (ClientNotFoundException e) {
```

Comment:
Is there any point throwing an exception here to immediately catch it? (In general, I'm kind of anti throwing exceptions although we've got a way to go in our codebase to get rid of them)

An alternative that I think sonar would be happy with:

```
        if (userContext.getClient().isEmpty()) {
            LOG.warn("Client not found");
            return generateApiGatewayProxyErrorResponse(400, ErrorResponse.ERROR_1015);
        }
        var client = userContext.getClient().get();
```